src/main/java/com/github/cameltooling/lsp/internal/diagnostic/DiagnosticService.java

```diff
@@ -40,7 +40,9 @@ protected DiagnosticService(CompletableFuture<CamelCatalog> camelCatalog) {
 	}
 
 	protected void logExceptionValidatingDocument(String docUri, Exception e) {
-		LOGGER.warn("Error while trying to validate the document {0}", docUri, e);
+		if (LOGGER.isWarnEnabled()) {
+			LOGGER.warn(String.format("Error while trying to validate the document %s", docUri), e);
+		}		
```

Comment:
worth double-checking but I think that a cleaner solution is

`LOGGER.warn("Error while trying to validate the document {}", docUri, e);`

Reply:
tested that it is working.
provided PR https://github.com/camel-tooling/camel-language-server/pull/525